### ISSUE TITLE
Dependencies upgrade

### ISSUE THREAD

**Issue by mdantonio** [rapydo/issues#130]
Prepared a simple script to extract dependencies from several files (Dockerfile, compose files, setup.py, requirements.txt, package.json) to obtain as output a list of dependecies (with specified fixed versions)

This is the current status:

```
{
    'postgres': {'compose': 'postgres:10.7-alpine'},
    'mariadb': {'compose': 'mariadb:10.4.4'},
    'neo4j': {'compose': 'neo4j:3.5.3'},
    'mongodb': {'compose': 'mongo:4.1.4'},
    'sqlalchemyui': {'compose': 'adminer:4.7.1-standalone'},
    'mongoui': {'compose': 'mongo-express:0.45.0'},
    'backend': {'Dockerfile': 'ubuntu:18.04', 'pip': ['tornado==5.1.1']},
    'icat': {'Dockerfile': 'ubuntu:17.10'},
    'proxy': {'Dockerfile': 'nginx:1.15.10-alpine'},
    'angular': {
        'package.json': [
            '@angular/common:7.2.7',
            '@angular/compiler:7.2.7',
            '@angular/core:7.2.7',
            '@angular/forms:7.2.7',
            '@angular/platform-browser:7.2.7',
            '@angular/platform-browser-dynamic:7.2.7',
            '@angular/router:7.2.7',
            '@angular/upgrade:7.2.7',
            'rxjs:6.4.0',
            'rxjs-compat:6.4.0',
            'bootstrap:4.3.1',
            '@ng-bootstrap/ng-bootstrap:4.0.4',
            'core-js:2.6.5',
            'zone.js:0.8.29',
            '@ngx-formly/core:4.8.2',
            '@ngx-formly/bootstrap:4.8.2',
            '@swimlane/ngx-datatable:14.0.0',
            'noty:3.2.0-beta',
            'moment:2.24.0',
            'ngx-moment:3.3.0',
            'angular-confirmation-popover:4.2.0',
            'ngx-clipboard:11.1.9',
            'ng2-file-upload:1.3.0',
            'file-saver:2.0.1',
            '@types/file-saver:2.0.0',
            '@fortawesome/fontawesome-free:5.7.2',
            'material-icons:0.3.0',
            'jquery:3.3.1',
            'typescript:3.3.3333',
            'webpack:4.29.6',
            'webpack-cli:3.2.3',
            'webpack-dev-server:3.2.1',
            'extract-text-webpack-plugin:4.0.0-beta.0',
            'html-webpack-plugin:3.2.0',
            'copy-webpack-plugin:5.0.0',
            'css-loader:2.1.0',
            'file-loader:3.0.1',
            'html-loader:0.5.5',
            'raw-loader:1.0.0',
            'style-loader:0.23.1',
            'null-loader:0.1.1',
            'webpack-merge:4.2.1',
            '@types/node:11.10.4',
            'angular2-template-loader:0.6.2',
            'awesome-typescript-loader:5.2.1',
            'rimraf:2.6.3'
        ]
    },
    'backendirods': {
        'Dockerfile': 'ubuntu:17.10',
        'pip': ['tornado==5.1.1', 'gssapi==1.5.0']
    },
    'utilities': ['PyYAML==3.13', 'sultan==0.8.1'],
    'controller': ['docker-compose==1.24.0', 'gitpython==2.1.11'],
    'http-api': [
        'flask==1.0.2',
        'requests-oauthlib==1.1.0',
        'injector==0.12',
        'flask_injector==0.10.1',
        'neomodel==3.3.1'
    ]
}
```

By removing all  up-to-date dependencies and versions very near to the latest (e.g. nginx:1.15.10-alpine is released 9 days ago, latest is nginx:1.15.12-alpine wth 1.15.10 released 2 days ago) we have the following deps that should be updated:

 ```
{
    'mongodb': {'compose': 'mongo:4.1.4'},
    'mongoui': {'compose': 'mongo-express:0.45.0'},
    'backend': {'Dockerfile': 'ubuntu:18.04', 'pip': ['tornado==5.1.1']},
    'icat': {'Dockerfile': 'ubuntu:17.10'},
    'backendirods': {
        'Dockerfile': 'ubuntu:17.10',
        'pip': ['tornado==5.1.1', 'gssapi==1.5.0']
    },
    'utilities': ['PyYAML==3.13', 'sultan==0.8.1'],
    'http-api': [
        'requests-oauthlib==1.1.0',
        'injector==0.12',
        'flask_injector==0.10.1',
    ]
}
```

**mongo:4.1.4** released: 19th Oct 2018; latest = 4.1.10 (12th Apr 2019)

**mongo-express:0.45.0** released: 17th Jul 2018; latest = 0.49 (109h Mar 2019)

(backend) **ubuntu:18.04** since it is an LTS we could avoid the upgrade to 18.10 or 19.04

(backendirods + icat) **ubuntu:17.10**: we should upgrade to 18.04LTS or 18.10 / 19.04 ... but probably install irods will be difficult

**tornado==5.1.1** released: 16th Sep 2018, latest = 6.0.2 (23th Mar 2019). This version has been fixed due to incompatibility of tornado (6.0.1) with flower, we should try to remove this dependency

**gssapi==1.5.0** released 6th Apr 2018, latest = 1.5.1 18th Jul 2018. I remember some compatility issue... to be retried

**PyYAML==3.13** released 6th Jul 2018, latest = 5.1 (released 13th Mar 2019). We cannot upgarde since compose 1.24.0 still require PyYAML 3.13

**sultan==0.8.1** released 5th Jul 2018, latest = 0.9.1 (released 4th Mar 2019). Already tried with compatibility errors, to be retried

**requests-oauthlib==1.1.0** released 9th Jan 2019, latest = 1.2.0 (released 14th Jan 2019)

**injector==0.12** released 9th Feb 2017, latest = 0.15 (released 7th Feb 2019). Probably compatibility issues.

**flask_injector==0.10.1** released 18th Oct 2017, latest = 0.12.0 (released 5th Apr 2019)

**Comment by mdantonio** [comment]
Upgraded neomodel from 3.2.9 to 3.3.1

This introduces a compatibility issue when the the User model is extended.

I found a workaround to fix the issue, by including the following code in the project neo4j models file, just before the User class redefinition:

```
from neomodel.util import NodeClassRegistry

registry = NodeClassRegistry()
base_user = frozenset({'User'})
for c in registry._NODE_CLASS_REGISTRY:
    if c == base_user:
        registry._NODE_CLASS_REGISTRY.pop(base_user)
        break
```

**Comment by mdantonio** [comment]
> injector==0.12 released 9th Feb 2017, latest = 0.15 (released 7th Feb 2019). Probably compatibility issues.
As reported before we are stuck at version 0.12 (two year old now) due to the following error:

> AssertionError: Passing keyword arguments to inject is no longer supported. Use inject in combination with parameter  annotations to declare dependencies

Before investigating more in details, I will try to replace injector with a similar lib (Inject)

**Comment by mdantonio** [comment]
> mongo:4.1.4 released: 19th Oct 2018; latest = 4.1.10 (12th Apr 2019)

No priority

> mongo-express:0.45.0 released: 17th Jul 2018; latest = 0.49 (109h Mar 2019)

No priority

> (backend) ubuntu:18.04 since it is an LTS we could avoid the upgrade to 18.10 or 19.04

Let's keep the LTS

> (backendirods + icat) ubuntu:17.10: we should upgrade to 18.04LTS or 18.10 / 19.04 ... but probably install irods will be difficult

No way to upgrade without requiring a lot of effort (main problem is GSI)

> tornado==5.1.1 released: 16th Sep 2018, latest = 6.0.2 (23th Mar 2019). This version has been fixed due to incompatibility of tornado (6.0.1) with flower, we should try to remove this dependency

Dropped this depedencies and introduced flower==0.9.3 (latest)

> gssapi==1.5.0 released 6th Apr 2018, latest = 1.5.1 18th Jul 2018. I remember some compatility issue... to be retried

Upgraded

> PyYAML==3.13 released 6th Jul 2018, latest = 5.1 (released 13th Mar 2019).

We cannot upgarde since compose 1.24.0 still require PyYAML 3.13

> sultan==0.8.1 released 5th Jul 2018, latest = 0.9.1 (released 4th Mar 2019). Already tried with compatibility errors, to be retried

Upgraded

> requests-oauthlib==1.1.0 released 9th Jan 2019, latest = 1.2.0 (released 14th Jan 2019)

We cannot upgrade. Version 1.2.0 depends on OAuthlib 3.0.0 and above, but Flask-OAuthlib depends from OAuthlib < 3.0.0

> injector==0.12 released 9th Feb 2017, latest = 0.15 (released 7th Feb 2019). Probably compatibility issues.

No way to upgrade. From 0.13 passing keyword arguments to inject is no longer supported and the current implementation is based on such feature. Tried to change the implementation => no success. Tried to replace the lib with injector => no success. Too many effort is required

> flask_injector==0.10.1 released 18th Oct 2017, latest = 0.12.0 (released 5th Apr 2019)

Compatibility issues, to be retried in the future
